magma/circuit.py

```diff
@@ -735,6 +735,13 @@ def __new__(metacls, name, bases, dct):
 
         return self
 
+    def __getattr__(self, attr):
+        try:
+            return only(filter(lambda i: i.name == attr, self.instances))
+        except IterableException:
+            pass
+        return object.__getattribute__(self, attr)
```

Comment:
Could we invert this, do `__getattribute__` by default, and check instances list on AttributeError? That would optimize the common case (save a level of indirection in the function call stack)

Reply:
I think the above behavior is expected given normal Python class/object semantics so I'd advocate for preserving that and having a get API specifically for instances when there is this issue of collisions.

Reply:
Good point. See #1262